src/analytics/analytics.tsx

```diff
@@ -0,0 +1,17 @@
+// import segment
+import * as DSL from "./dsl"
+
+export function sendIntent(analyticsConstant: string, contextProps: any) {
+  // send it
+  let properties = DSL.dsl[analyticsConstant].intent(contextProps)
```

Comment:
As @alloy mentioned earlier, if we store handlers in a map, we can check the constant was registered properly.
```
const handlers = new Map()
if(handlers.has(analyticsConstant)) {
  handlers.get(analyticsConstant).intent(contextProps)
}
```